rcl/test/rcl/test_arguments.cpp

```diff
@@ -1099,3 +1181,48 @@ TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), test_bad_allocs_copy
   }
   parsed_args.impl->allocator = saved_alloc;
 }
+
+TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), test_null_get_param_files) {
+  const std::string parameters_filepath1 = (test_path / "test_parameters.1.yaml").string();
+  const char * const argv[] = {
+    "process_name", "--ros-args", "--params-file", parameters_filepath1.c_str()
+  };
+  const int argc = sizeof(argv) / sizeof(const char *);
+
+  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
+  char ** parameter_files = NULL;
+  rcl_allocator_t allocator = rcl_get_default_allocator();
+  rcl_ret_t ret = rcl_parse_arguments(argc, argv, allocator, &parsed_args);
+  ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+  OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT(
+  {
+    EXPECT_EQ(RCL_RET_OK, rcl_arguments_fini(&parsed_args));
+  });
+
+  ret = rcl_arguments_get_param_files(nullptr, allocator, &parameter_files);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string().str;
+
+  ret = rcl_arguments_get_param_files(&parsed_args, allocator, nullptr);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string().str;
+
+  rcl_arguments_t empty_parsed_args = rcl_get_zero_initialized_arguments();
+  ret = rcl_arguments_get_param_files(&empty_parsed_args, allocator, &parameter_files);
+  EXPECT_EQ(RCL_RET_INVALID_ARGUMENT, ret) << rcl_get_error_string().str;
+}
+
+TEST_F(CLASSNAME(TestArgumentsFixture, RMW_IMPLEMENTATION), test_bad_alloc_parse_arg) {
+  const std::string parameters_filepath1 = (test_path / "test_parameters.1.yaml").string();
+  const char * const argv[] = {
+    "process_name", "--ros-args", "--params-file", parameters_filepath1.c_str()
+  };
+  const int argc = sizeof(argv) / sizeof(const char *);
+
+  rcl_arguments_t parsed_args = rcl_get_zero_initialized_arguments();
+  rcl_allocator_t bomb_alloc = get_time_bombed_allocator();
+
+  for (int i = 0; i < 9; i++) {
+    set_time_bombed_allocator_count(bomb_alloc, i);
+    rcl_ret_t ret = rcl_parse_arguments(argc, argv, bomb_alloc, &parsed_args);
+    EXPECT_EQ(RCL_RET_BAD_ALLOC, ret) << rcl_get_error_string().str;
+  }
```

Comment:
@Blast545 meta: hmm, I wonder if this isn't better suited for fault injection. That magic `9` is a bit fragile.

Reply:
I am working on a followup PR to rcutils that will include a fault_injection_allocator that accomplishes similar functionality to the time_bomb_allocator. If you think that would be useful to this PR, I can prioritize that.

Reply:
@hidmic I agree it's somewhat fragile. In fact I was considering some sort of macro for testing functions that received allocators to iterate until the test passed. Although probably the fault injection approach will be better in the long run, and that's why I left the fixed number for this PR.

@brawner I'm not fully familiarized yet with the fault_injection tools you are working at the moment. Although I imagined the injection tools to be included in the `default_allocator` used through the ROS 2 code to be able of testing bad allocator fails like this one.

We can always merge this PR and updated it with injection tools once these are ready, WDYT @brawner @hidmic ?

Reply:
> Although I imagined the injection tools to be included in the default_allocator

That is probably a better approach. While most C code requires passing in an allocator, I think some c++ code just uses the default allocator blindly.

To make the test a little less fragile, you can do something like:

```c++
  for (int i = 0; i < 9; i++) {
    set_time_bombed_allocator_count(bomb_alloc, i);
    rcl_ret_t ret = rcl_parse_arguments(argc, argv, bomb_alloc, &parsed_args);
    if (RCL_RET_OK == ret) {
      EXPECT_EQ(RCL_RET_OK, rcl_arguments_fini(&parsed_args));
    } else {
      EXPECT_EQ(RCL_RET_BAD_ALLOC, ret);
      EXPECT_TRUE(rcl_error_is_set());  // I don't know if this is true
      rcl_error_reset();
    }
  }
```

Reply:
Err, edited to change the 9 in for loop to something large, like 100.

Reply:
With the missing break, right?
```c++
  for (int i = 0; i < 100; i++) {
    set_time_bombed_allocator_count(bomb_alloc, i);
    rcl_ret_t ret = rcl_parse_arguments(argc, argv, bomb_alloc, &parsed_args);
    if (RCL_RET_OK == ret) {
      EXPECT_EQ(RCL_RET_OK, rcl_arguments_fini(&parsed_args));
      break;
    } else {
      EXPECT_EQ(RCL_RET_BAD_ALLOC, ret);
      EXPECT_TRUE(rcl_error_is_set());
      rcl_error_reset();
    }
  }
```

Reply:
Oooh, good idea. I like it

Reply:
Done 2ac0015